miniscope_io/device_update.py

```diff
@@ -1,144 +1,105 @@
-        if module == "LED":
-            assert value <= ledMAX and value >= ledMIN
-        if module == "EWL":
-            assert value <= ewlMAX and value >= ewlMIN
-    except AssertionError as msg:
-        if module == "LED":
-            logger.exception("LED value need to be a integer within 0-100")
-        if module == "EWL":
-            logger.exception("EWL value need to be an integer within 0-255")
-        raise msg
-
-    if module == "LED":
-        deviceTag = ledDeviceTag << deviceTagPos
-    elif module == "EWL":
-        deviceTag = ewlDeviceTag << deviceTagPos
-
-    command = [0, 0]
-
-    command[0] = int(
-        Preamble[0] * 2**preamblePos + deviceTag + np.floor(value / (2**uartPayload))
-    ).to_bytes(1, "big")
-    command[1] = int(
-        Preamble[1] * 2**preamblePos + deviceTag + value % (2**uartPayload)
-    ).to_bytes(1, "big")
-
-    # set up serial port
-    try:
-        serial_port = serial.Serial(port=comport, baudrate=baudrate, timeout=5, stopbits=1)
+        serial_port = serial.Serial(port=command.port, baudrate=2400, timeout=5, stopbits=2)
     except Exception as e:
         logger.exception(e)
         raise e
     logger.info("Open serial port")
 
-    for uartCommand in command:
-        for _ in range(uartRepeat):
-            # read UART data until preamble and put into queue
-            serial_port.write(uartCommand)
-            time.sleep(uartTimeGap)
+    try:
+        id_command = (command.device_id + UpdateCommandDefinitions.id_header) & 0xFF
+        serial_port.write(id_command.to_bytes(1, "big"))
+        logger.debug(f"Command: {format(id_command, '08b')}; Device ID: {command.device_id}")
+        time.sleep(0.1)
+
+        target_command = (command.target.value + UpdateCommandDefinitions.target_header) & 0xFF
+        serial_port.write(target_command.to_bytes(1, "big"))
+        logger.debug(f"Command: {format(target_command, '08b')}; Target: {command.target.name}")
+        time.sleep(0.1)
+
+        value_LSB_command = (
+            (command.value & UpdateCommandDefinitions.LSB_value_mask)
+            + UpdateCommandDefinitions.LSB_header
+        ) & 0xFF
+        serial_port.write(value_LSB_command.to_bytes(1, "big"))
+        logger.debug(f"Command: {format(value_LSB_command, '08b')}; Value: {command.value} (LSB)")
+        time.sleep(0.1)
+
+        value_MSB_command = (
+            ((command.value & UpdateCommandDefinitions.MSB_value_mask) >> 6)
+            + UpdateCommandDefinitions.MSB_header
+        ) & 0xFF
+        serial_port.write(value_MSB_command.to_bytes(1, "big"))
+        logger.debug(f"Command: {format(value_MSB_command, '08b')}; Value: {command.value} (MSB)")
+        time.sleep(0.1)
+
+        serial_port.write(UpdateCommandDefinitions.reset_byte.to_bytes(1, "big"))
+
+    finally:
+        serial_port.close()
+        logger.info("Closed serial port")
+
+
+def find_ftdi_device() -> list:
+    """
+    Find FTDI devices connected to the computer.
+    """
```

Comment:
```suggestion
    """
    Find FTDI devices connected to the computer.
    
    Returns:
        list[str]: list of matching device paths, e.g. `/dev/ttyUSB0` from :attr:`~pyserial.tools.list_ports.ListPortInfo`
    """
```

Reply:
just pushed a commit to add pyserial to the intersphinx mapping so that this works.